src/main/java/com/conveyal/datatools/manager/controllers/api/GtfsPlusController.java

```diff
@@ -144,7 +158,7 @@ private static HttpServletResponse downloadGtfsPlusFile(File file, Response res)
             bufferedOutputStream.flush();
             bufferedOutputStream.close();
         } catch (Exception e) {
-            halt(500, "Error serving GTFS+ file");
+            haltWithMessage(req, 500, "an unexpected error occurred", e);
```

Comment:
Side note: we should also be logging the exception server side here.

Reply:
Whenever I saw a blanket Exception catcher I used this generic unexpected error.  If there is additional logic added within this try/catch block the exception could end up being a range of different things.  Ideally, we should be doing more checking using more specific exception types to determine the exact cause of the exception.  We could then report the exact problem back to the user.  If we did that we may find that we could report more HTTP 400 errors indicating bad user input other than a 500 error which indicates a problem with the server's code.

Reply:
For now, can you add a catch clause for `IOException` here (and the other 500 error instance in this file) and have it halt with 400 and the previous error message.  Since we haven't fully tested this, I think we should still keep the blanket Exception catcher though and use your message with the 500 error code.  Is that approach suitable?

Reply:
Outdated comment, marking as resolved.